src/utils/general.js

```diff
@@ -0,0 +1,14 @@
+export const getStrippedText = (text, stripLength) => {
+  return text.length > stripLength
+    ? `${text.substring(0, stripLength)}...`
+    : text;
+};
+
+export const delay = (ms) => new Promise((res) => setTimeout(res, ms));
+
+export const formatNumber = (num) => {
+  if (num < 1e3) return num;
+  else if (num >= 1e3 && num < 1e6) return (num / 1e3).toFixed(1) + "K";
+  else if (num >= 1e6 && num < 1e9) return (num / 1e6).toFixed(1) + "M";
+  else if (num >= 1e9 && num < 1e12) return (num / 1e9).toFixed(1) + "B";
```

Comment:
what is `1e3` `1e6` and `1e9`? also `K, M, B` ?

Reply:
This is to add the necessary suffix for the views count. 10000 => 10k. 1e3 is 10 to the power of 3